src/InMemoryDocumentStore.php

```diff
@@ -94,7 +105,11 @@ public function hasCollectionIndex(string $collectionName, string $indexName): b
      */
     public function addCollectionIndex(string $collectionName, Index $index): void
     {
-        //InMemoryDocumentStore ignores indices
+        if($index instanceof FieldIndex || $index instanceof MultiFieldIndex) {
+            $this->dropCollectionIndex($collectionName, $index->name());
+        }
+
+        $this->inMemoryConnection['documentIndices'][] = $index;
```

Comment:
When an unique index is added and the store already contains documents, the store should ensure that unique constrainsts are not violated.

Reply:
ah of course, I missed that case. Thx for the pointer @jsor. I'll prepare a bugfix release